### PR TITLE
Devtooling 194

### PR DESCRIPTION
**Pull request by kavinbalagen**
Merge Request for linebase and phonebase

### CODE REVIEW

**Review by charliecon — APPROVED** [review]

File: genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+		if getErr != nil {
+			if gcloud.IsStatus404(resp) {
+				return retry.RetryableError(fmt.Errorf("Failed to read phone base settings %s: %s", d.Id(), getErr))
+			}
+			return retry.NonRetryableError(fmt.Errorf("Failed to read phone base settings %s: %s", d.Id(), getErr))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourcePhoneBaseSettings())
+		d.Set("name", *phoneBaseSettings.Name)
+		if phoneBaseSettings.Description != nil {
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go:125]: You can replace these if statements with the functions from the `resourcedata` package

File: genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	"github.com/mypurecloud/platform-client-sdk-go/v116/platformclientv2"
+)
+
+func createPhoneBaseSettings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	description := d.Get("description").(string)
+	phoneMetaBase := gcloud.BuildSdkDomainEntityRef(d, "phone_meta_base_id")
+	properties := gcloud.BuildBaseSettingsProperties(d)
+
+	phoneBase := platformclientv2.Phonebase{
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go:28]: Extract this logic into to build the phonebase object into its own function, the same logic is repeated in the update function

File: genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+	if description != "" {
+		phoneBase.Description = &description
+	}
+
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
+
+	log.Printf("Creating phone base settings %s", name)
+	phoneBaseSettings, _, err := edgesAPI.PostTelephonyProvidersEdgesPhonebasesettings(phoneBase)
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go:49]: Why is there no proxy file?

File: genesyscloud/telephony_providers_edges_linebasesettings/data_source_genesyscloud_telephony_providers_edges_linebasesettings.go [hunk 4]
```diff
@@ -1,38 +1,26 @@
...
+	sdkConfig := m.(*gcloud.ProviderMeta).ClientConfig
 	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
 
 	name := d.Get("name").(string)
 
-	return WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
+	return gcloud.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
 		for pageNum := 1; ; pageNum++ {
 			const pageSize = 50
 			lineBaseSettings, _, getErr := edgesAPI.GetTelephonyProvidersEdgesLinebasesettings(pageNum, pageSize, "", "", nil)
```
Comment (dginty4) [inline genesyscloud/telephony_providers_edges_linebasesettings/data_source_genesyscloud_telephony_providers_edges_linebasesettings.go:26]: Same here, why aren't we using a proxy?

**Pull request merged**